lib/LedgerSMB/Upgrade_Tests.pm

```diff
@@ -214,13 +231,37 @@ Tooltip for each button
 =cut
 
 has tooltips => (is => 'ro',
-    isa => 'HashRef[Str]',
-    default => sub {
-        return {
-            'Save and Retry' => marktext('Save the fixes provided and attempt to continue migration'),
-            'Cancel' => marktext('Cancel the <b>migration</b>')
-    }},
-    required => 0);
+    isa => 'Maybe[HashRef[Str]]', required => 0,
+    default => undef,   # Force initializer call
+    initializer => sub {
+        my ( $self, $value, $writer_sub_ref, $attribute_meta ) = @_;
+        for ($value) {
+            die LedgerSMB::I18N::text(q(No button '[_1]' in test '[_2]'),$_,$self->{name})
+                if not grep( /^$_$/, @{$self->{buttons}});
+        }
+        my %defaults = ('Save and Retry' => marktext('Save the fixes provided and attempt to continue migration'),
+                                'Cancel' => marktext('Cancel the <b>whole migration</b>'));
+        for (keys %defaults) {
+            $value->{$_} = $defaults{$_}
+                if not $value->{$_};
+        }
+        $writer_sub_ref->($value);
+    }
+);
+
+=item skip
+
+Skip this test
+
+=cut
+
+enum SkipValues => [qw/ Disabled Off On /];
+has skip => (is =>'ro', isa => 'SkipValues',
```

Comment:
This doesn't feel right: why do we have a special attribute to determine the status of exactly one of possibly infinite buttons?

Reply:
Tests can be skippable. Simplified and renamed